compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -503,6 +503,25 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     }
                 }
 
+                let attrs = self.tcx.codegen_fn_attrs(instance.def_id());
+                let missing_features: Vec<_> = attrs
+                    .target_features
+                    .iter()
+                    .copied()
+                    .filter(|feature| !self.tcx.sess.target_features.contains(feature))
+                    .collect();
```

Comment:
I get measurable perf changes when avoiding allocations in this function. So we should not create a new `Vec` here each time. Why do we need to manifest this list anyway? We just need to iterate over it and check if any of them is not in `target_features`.

Reply:
I did not expect such impact from `collect`ing into an empty `Vec`, so I will avoid it.